### PR TITLE
STARSolo Training

### PR DESCRIPTION
**Pull request by mtekman**
This adds a 10X based workflow

### CODE REVIEW

File: topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+>
+> * `nNinBarcode` + `nUMIhomopolymer` + `nNoMatch` + `nTooMany` + `nNoExactMatch` = number of reads with CBs that do not match whitelist.
+> * `nUnmapped` + `nAmbigFeature` = number of reads without defined feature (gene)
+> * `nMatch` = number of reads that are output as solo counts
+>
+> The three categoties above summed together should be equal to the total number of reads.
+>
+{: .details}
+
+The main information to gather at this stage is that the `nCellBarcodes` tell us how many cells were detected in our sample, where we see 272 which is expected of our sub-sampled data. Another metric to take into account is that the number of matches (`nMatch`) has the largest value, and that the number of reads that map to the genome but not to a feature/gene given in the GTF (`nNoFeature`) is not too large. The number of no features is also quite high when mapping the source files, so this appears to be the default expected behaviour.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md]: the term "source files" is confusing to me, can you call them original files from the paper?

File: topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+{: .hands_on}
+
+
+# Producing a Quality Count Matrix
+
+The matrix files produced by **RNA STARsolo** are in the *bundled* format, meaning that the information to create a tabular matrix of Genes vs Cells are separated into different files. These files are already 10x analysis datasets, compatible with any downstream single-cell RNA analysis pipeline, however the number of cells represented here are greatly over-represented, as they have not yet been filtered for high quality cells, and therefore the matrix represents *any* cells that were unambiguously detected in the sample.
+
+If we were to construct a cell matrix using the data we have, we would have a large matrix of 60,000 Genes against 737,000 Cells, of which most values would be zero, i.e. an *extremely* sparse matrix.
+
+To get a high quality count matrix we must apply the **DropletUtils** tool, which will produce a filtered set which will produce a dataset more representative of the CellRanger pipeline.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md]: wihch will produce ... which will produce ...?

I also feel you should somewhere mention what a CellRanger pipeline is if you cite it so much.

File: topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+>
+> > ### {% icon solution %} Solution
+> >
+> > 1. By clicking on the title of the output dataset in the history we can expand the box to see the output says that there are `272` cells in the output table.
+> > 1. If we expand the {% icon galaxy-eye %} *RNA STARsolo Feature Statistic Summaries* Dataset and look at the `nCellBarcodes` value, we see that this is also `272` cells, as expected.
+> >
+> {: .solution}
+{: .question}
+
+This will produce a count matrix in a human readable tabular format which can be used in downstream analysis tools, such as those used in the [*Downstream Single-cell RNA analysis with RaceID*](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-raceid/tutorial.html) tutorial.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md]: the link should be adopted to the `{{ site.baseurl }}{% link` format.

File: topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+>    - *"Format for the input matrix"*: `Bundled (barcodes.tsv, genes.tsv, matrix.mtx)`
+>        - {% icon param-file %} *"Count Data"*: `output_matrix` (output of **RNA STARsolo** {% icon tool %})
+>        - {% icon param-file %} *"Genes List"*: `output_genes` (output of **RNA STARsolo** {% icon tool %})
+>        - {% icon param-file %} *"Barcodes List"*: `output_barcodes` (output of **RNA STARsolo** {% icon tool %})
+>    - *"Operation"*: `Rank Barcodes`
+>        - *"Method"*: `EmptyDrops`
+>            - *"Lower Bound"*: `1`
+>
+>    > ### {% icon comment %} Comment
+>    > Due to the sub-sampled dataset, we choose a very low lower bound.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md]: Which number is more real?

File: topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+>    > The False Discovery Rate (FDR) is set to zero because the ambient profile is too weak to measure against on this dataset.
+>    >
+>    {: .comment}
+>
+{: .hands_on}
+
+![cells]({{ site.baseurl }}{% link topics/transcriptomics/images/tenx_cells.png %} "Detected Cells (red)")
+
+
+Here we recover 137 high quality cells instead of the detected 272 given previously. On large datasets, this difference can help clean downstream clustering which may appear softer and less well-defined on a more inclusive dataset than that which has undergone tighter quality controls.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md]: Can we split the second sentence into two and make the sentence easier to understand and therefore to translate?

File: topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+
+# Conclusion
+{:.no_toc}
+
+In this workflow we have learned to quickly perform mapping and quantification of scRNA-seq FASTQ data in a single step via **RNA STARsolo**, and have reproduced a CellRanger workflow using the **DropletUtils** suite, where we further explored the use of barcode rankings to determine better filtering thresholds to generate a high quality count matrix.
+
+A full pipeline which produces both an hda5 and tabular file for inspection is provided [here](workflows/scrna_tenx.ga).
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.md]: hda5? Please explain what that is ...
one of your questions is ` "What is an HDF5 file, and why is it important?"` but this part is missing.

**Comment by mtekman** [comment]
thanks for the review @bgruening, I've implemented your suggestions in the second version

**Comment by mtekman** [comment]
More reviews would also be welcome, otherwise hooray

**Comment by bgruening** [comment]
I'm sure we will iterate over this during the next weeks, but this is a very nice first version. Thanks @mtekman!

**Pull request merged**